### PR TITLE
Customizable gameOver listener and more

### PR DESCRIPTION
**Pull request by Darkness6030**
Why i did this?
Because i want a custom gameOver listener in my plugin. There are no ways to remove default one. But now you can change it to anything.

Why making public so many variables?
Because i am **very** tired of using Reflect.
Note that i made public only useful variables like nextMapOverride.

Also i moved 2 final variabled to Vars.java. Now you can use it anywhere.

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by LeoSko** [comment]
Another thought: events map from Events from arc also should be public, or provide less restricted interface of removing listeners.

File: core/src/mindustry/Vars.java [hunk 1]
```diff
@@ -141,6 +141,10 @@
     public static final int multicastPort = 20151;
     /** multicast group for discovery.*/
     public static final String multicastGroup = "227.2.7.7";
+    /** extra time after the end of the game. */
+    public static final int roundExtraTime = 12;
```
Comment (Anuken) [inline core/src/mindustry/Vars.java:145]: What is the point of placing either of these variables here? They're still static final, and they have nothing to do with "shared" variables. Either make them `Config` numbers, or don't move them.

File: server/src/mindustry/server/ServerControl.java [hunk 2]
```diff
@@ -35,16 +36,18 @@
...
-
     protected static String[] tags = {"&lc&fb[D]&fr", "&lb&fb[I]&fr", "&ly&fb[W]&fr", "&lr&fb[E]", ""};
     protected static DateTimeFormatter dateTime = DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm:ss"),
         autosaveDate = DateTimeFormatter.ofPattern("MM-dd-yyyy_HH-mm-ss");
 
     public final CommandHandler handler = new CommandHandler("");
     public final Fi logFolder = Core.settings.getDataDirectory().child("logs/");
 
+    public boolean inExtraRound;
+    public Task lastTask;
```
Comment (Anuken) [inline server/src/mindustry/server/ServerControl.java:47]: Implementation detail, confusing and should not be public. What is a "last task"? What does it refer to? Can it be null? What would canceling it do? When does accessing it change anything at all? None of these are clear from the variable alone.

File: server/src/mindustry/server/ServerControl.java [hunk 3]
```diff
@@ -35,16 +36,18 @@
...
+    public Gamemode lastMode;
+    public @Nullable Map nextMapOverride;
```
Comment (Anuken) [inline server/src/mindustry/server/ServerControl.java:49]: Should be private and set with a setter to support future changes.

File: server/src/mindustry/server/ServerControl.java [hunk 4]
```diff
@@ -35,16 +36,18 @@
...
     protected static String[] tags = {"&lc&fb[D]&fr", "&lb&fb[I]&fr", "&ly&fb[W]&fr", "&lr&fb[E]", ""};
     protected static DateTimeFormatter dateTime = DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm:ss"),
         autosaveDate = DateTimeFormatter.ofPattern("MM-dd-yyyy_HH-mm-ss");
 
     public final CommandHandler handler = new CommandHandler("");
     public final Fi logFolder = Core.settings.getDataDirectory().child("logs/");
 
+    public boolean inExtraRound;
+    public Task lastTask;
+    public Gamemode lastMode;
```
Comment (Anuken) [inline server/src/mindustry/server/ServerControl.java:48]: Should use a different name and/or documented getter to let users know what this really stores.

**Review by Anuken — CHANGES_REQUESTED** [review]

**Comment by way-zer** [comment]
I do this to disable default gameover logic
Snipaste_2022-03-03_10-19-25
https://github.com/way-zer/ScriptAgent4MindustryExt/blob/4fb49cc79b53c5915da205a43e36c19b88166f9f/scripts/wayzer/maps.kts#L86

**Pull request closed without merging**